src/community/features-templating/features-templating-web/src/main/java/org/geoserver/featurestemplating/web/TemplateModelsValidator.java

```diff
@@ -0,0 +1,64 @@
+/* (c) 2021 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.featurestemplating.web;
+
+import org.geoserver.featurestemplating.configuration.SupportedFormat;
+import org.geoserver.featurestemplating.configuration.TemplateInfo;
+import org.geoserver.featurestemplating.configuration.TemplateRule;
+import org.geoserver.featurestemplating.expressions.MimeTypeFunction;
+
+public class TemplateModelsValidator {
+
+    public void validate(TemplateInfo info, String rawTemplate)
+            throws TemplateConfigurationException {
+        String templateName = info.getTemplateName();
```

Comment:
Kept those methods only for Preview object and rule object to avoid the form submitting to complain when actually the form should be considered valid.
For the TemplateInfo form I made them mandatory at compoment level as suggested.